erpnext/accounts/doctype/payment_reconciliation/payment_reconciliation.js

```diff
@@ -120,13 +120,38 @@ erpnext.accounts.PaymentReconciliationController = class PaymentReconciliationCo
 			doc: me.frm.doc,
 			method: 'get_unreconciled_entries',
 			callback: function(r, rt) {
-				me.set_invoice_options();
 				me.toggle_primary_action();
 			}
 		});
 
 	}
 
+	allocate() {
+		var me = this;
+		let payments = me.frm.fields_dict.payments.grid.get_selected_children();
+		if (!(payments.length)) {
+			payments = me.frm.doc.payments;
+		}
+		let invoices = me.frm.fields_dict.invoices.grid.get_selected_children();
+		if (!(invoices.length)) {
+			invoices = me.frm.doc.invoices;
+		}
+		return me.frm.call({
+			doc: me.frm.doc,
+			method: 'allocate_entries',
+			args: {
+				payments: payments,
+				invoices: invoices
+			},
+			callback: function(r) {
+				if (!r.exc && r.message) {
+					me.frm.set_value("allocation", r.message);
+					me.toggle_primary_action();
+				}
+			}
+		});
+	}
+
 	reconcile() {
 		var me = this;
 		var show_dialog = me.frm.doc.payments.filter(d => d.difference_amount && !d.difference_account);
```

Comment:
Payments table doesn't have `difference_amount` & `difference_account` field now. You need to handle it with allocation table